rustls/src/server/verifier_builder.rs

```diff
@@ -0,0 +1,427 @@
+    pub fn build(self) -> Arc<dyn ClientCertVerifier> {
+        Arc::new(AllowAnyAuthenticatedClient::new(
+            self.state.roots,
+            Vec::default(),
+        ))
+    }
+
+    /// Build a client certificate that will allow clients that present a client certificate
+    /// that is issued by a trusted CA, or anonymous clients that do not offer a client certificate
+    /// at all. No revocation checking with CRLS will be done.
+    pub fn allow_unauthenticated(self) -> ClientCertVerifierBuilder<ClientAuthenticationOptional> {
+        ClientCertVerifierBuilder {
+            state: ClientAuthenticationOptional {
+                roots: self.state.roots,
+                crls: Vec::default(),
+            },
+        }
+    }
+
+    /// Build a client certificate verifier that will require all clients present a client
+    /// certificate that is issued by a trusted CA. No revocation checking with CRLs will be done.
+    /// No anonymous clients will be allowed to access the server.
+    ///
+    /// Equivalent to calling [build()][ClientCertVerifierBuilder<WantsCrls>::build].
+    pub fn without_crls(self) -> ClientCertVerifierBuilder<WantsUnauthenticatedPolicy> {
+        ClientCertVerifierBuilder {
+            state: WantsUnauthenticatedPolicy {
+                roots: self.state.roots,
+                crls: Vec::default(),
+            },
+        }
+    }
+
+    /// Build a client certificate verifier that will require all clients present a client
+    /// certificate that is issued by a trusted CA. The revocation state of the client
+    /// certificate will be checked against the provided certificate revocation lists (CRLs).
+    ///
+    /// All clients must provide a client certificate unless you further allow unauthenticated
+    /// clients by calling
+    /// [allow_unauthenticated()][ClientCertVerifierBuilder<WantsUnauthenticatedPolicy>::allow_unauthenticated]
+    /// on the returned builder.
+    pub fn with_crls(
+        self,
+        crls: impl IntoIterator<Item = UnparsedCertRevocationList>,
+    ) -> Result<ClientCertVerifierBuilder<WantsUnauthenticatedPolicy>, CertRevocationListError>
+    {
+        let crls = crls
+            .into_iter()
+            .map(|der_crl| der_crl.parse())
+            .collect::<Result<Vec<_>, CertRevocationListError>>()?;
+
+        Ok(ClientCertVerifierBuilder {
+            state: WantsUnauthenticatedPolicy {
+                roots: self.state.roots,
+                crls,
+            },
+        })
+    }
+}
+
+/// A client certificate verifier builder state where clients are verified using a [RootCertStore]
+/// and optionally may have their revocation status checked via CRLs. The caller must now decide
+/// or not to allow unauthenticated clients in addition to those that supply a valid client
+/// certificate.
+///
+/// For more information, see the [`WebpkiClientVerifier`] documentation.
+// TODO(XXX): derive Clone once OwnedCertRevocationList does.
```

Comment:
I will follow up on this in `webpki` but didn't want to block the builder API on it.

Reply:
https://github.com/rustls/webpki/pull/129

Reply:
We're on the updated webpki in `main` so I was able to resolve this.